frontend/src/components/professor/ChapterDetails/ChapterDetails.js

```diff
@@ -83,21 +84,36 @@ function ChapterDetails(props) {
     getChapterDetails()
   }, [getChapterDetails])
 
+  useEffect(() => {
+    if (chosenActivityData?.jsonConfig) {
+      setIsEditActivityModalOpen(true)
+    }
+  }, [chosenActivityData?.jsonConfig])
+
+  const getActivityInfo = useCallback((activityId) => {
+    ActivityService.getActivityInfo(activityId)
+      .then((response) => {
+        return setChosenActivityData((prevState) => ({ ...prevState, jsonConfig: response.activityBody }))
+      })
+      .catch(() => {
+        setChosenActivityData((prevState) => ({ ...prevState, jsonConfig: null }))
+      })
+  }, [])
+
   const goToChapterDetails = (activityName, activityId, activityType) => {
     navigate(location.pathname + `/activity/${activityName}`, {
       state: { activityId: activityId, activityType: activityType }
     })
   }
 
   const startActivityEdition = (activity) => {
-    // TODO: depending on the type of activity, we will use a different endpoint
+    setReloadMapNeeded(false)
```

Comment:
Ten setter dałem, żeby mapa się przeładowała po drugiej i kolejnej edycji jakiejś aktywności bez przeładowania po drodze. Bo gdy zedytujemy aktywność po raz pierwszy to wartość dla tej zmiennej zmieni się na true i wyzwoli reload. Ale przy kolejnym zmiana z true na true nie wyzwala useEffecta w reloadzie